### PR TITLE
added keyboard key binding to buttons functionality

### PR DESCRIPTION
**Pull request by mohamedar97**
Fixes #
added the functionality requested in issue #200 kindly try the code and give my feedback
#### Describe the changes you have made in this pr -
added 2 properties to button module
added a keyup and a keydown events
added an input to field to the properties list in the UX.js file and attached an event to this input field

### CODE REVIEW

File: public/js/listeners.js [hunk 1]
```diff
@@ -2,7 +2,18 @@
 
 
 function startListeners() {
-    window.addEventListener('keyup', function(e) {
+    window.addEventListener('keyup', function (e) {
+        if (globalScope.Button.length > 0) {
```
[the remark this answers is not in the document]
Comment (mohamedar97) [reply public/js/listeners.js]: i double checked this and i cant see that duplication in my code

File: public/js/listeners.js [hunk 2]
```diff
@@ -38,19 +49,31 @@ function startListeners() {
...
 
 
-    window.addEventListener('keydown', function(e) {
+    window.addEventListener('keydown', function (e) {
 
         // If mouse is focusing on input element, then override any action
         // if($(':focus').length){
         //     return;
         // }
+        if (globalScope.Button.length > 0) {
```
[the remark this answers is not in the document]
Comment (mohamedar97) [reply public/js/listeners.js]: the duplication is because i used almost the same code in different events

File: public/js/listeners.js [hunk 3]
```diff
@@ -2,7 +2,18 @@
 
 
 function startListeners() {
-    window.addEventListener('keyup', function(e) {
+    window.addEventListener('keyup', function (e) {
+        if (globalScope.Button.length > 0) {
+
+            globalScope.Button.forEach((button, index) => {
+                if (e.key == button.activationKey) {
+                    button.wasClicked = false
```
[the remark this answers is not in the document]
Comment (mohamedar97) [reply public/js/listeners.js]: i honestly don't understand this error

File: public/js/listeners.js [hunk 4]
```diff
@@ -2,7 +2,18 @@
 
 
 function startListeners() {
-    window.addEventListener('keyup', function(e) {
+    window.addEventListener('keyup', function (e) {
+        if (globalScope.Button.length > 0) {
+
+            globalScope.Button.forEach((button, index) => {
```
[the remark this answers is not in the document]
Comment (mohamedar97) [reply public/js/listeners.js]: that was for debugging purposes. will be removed

**Comment by mohamedar97** [comment]
the error with the check of circleci is because postgres is added twice in the gem file and because i didnt use bundler 2.0.

**Comment by satu0king** [comment]
I think you missed some files. I cannot see modules.js

**Comment by mohamedar97** [comment]
@satu0king  the only two changes i made to modules.js are in lines 2417 & 2418 if you look closely you'll see them in the changes i made

File: database.example.yml [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+# Do not set this db to the same as development or production.
+test:
+  <<: *default
+  database: circuitverse_test
+
+production:
+  username: <%= ENV['DB_USERNAME']%>
+  password: <%= ENV['DB_PASSWORD']%>
+  host: localhost
+  database: circuitverse_production
```
Comment (satu0king) [inline database.example.yml]: Why is this file added?

**Comment by satu0king** [comment]
Don't worry about codeclimate issues in javascript. Resolve the conflicts

File: database.example.yml (hunk 5 above)
Comment (mohamedar97) [reply database.example.yml]: i deleted the database.example.yml file and added it before i deploy the file

Comment (mohamedar97) [reply database.example.yml]: i added the database.example.yml because i deleted it by mistake

**Comment by satu0king** [comment]
Please resolve the conflicts, it might be better to send a new PR instead, to make things easier.
